tests/testsuite/git.rs

```diff
@@ -1828,6 +1845,734 @@ fn fetch_downloads() {
     p.cargo("fetch").with_stdout("").run();
 }
 
+#[cargo_test]
+fn gitoxide_clones_registry_with_shallow_protocol_and_follow_up_with_git2_fetch(
```

Comment:
Should we move shallow clone tests to their own file? It grows way too many 😆.

Reply:
A good point :D. They have been moved to `git_shallow.rs` in 8b3508cad78d33976d4ff62ef736a340ce8b0268.